Apps/GatewayApi/src/Services/DelegationService.cs

```diff
@@ -16,57 +16,103 @@
+        /// <param name="dataProtectionProvider">The injected data protection provider.</param>
         /// <param name="delegationDelegate">The injected delegation delegate.</param>
         /// <param name="hashDelegate">The injected hash delegate.</param>
+        /// <param name="emailQueueService">The injected email queue service.</param>
         /// <param name="patientDetailsService">The injected patient details service.</param>
         public DelegationService(
             IConfiguration configuration,
             IMapper autoMapper,
             ILogger<DelegationService> logger,
+            IDataProtectionProvider dataProtectionProvider,
             IDelegationDelegate delegationDelegate,
             IHashDelegate hashDelegate,
+            IEmailQueueService emailQueueService,
             IPatientDetailsService patientDetailsService)
         {
             this.configuration = configuration;
             this.autoMapper = autoMapper;
             this.logger = logger;
+            this.dataProtector = dataProtectionProvider.CreateProtector(DataProtectionPurpose);
             this.delegationDelegate = delegationDelegate;
             this.hashDelegate = hashDelegate;
+            this.emailQueueService = emailQueueService;
             this.patientDetailsService = patientDetailsService;
+            this.emailTemplateConfig = configuration.GetSection(EmailTemplateConfig.ConfigurationSectionKey).Get<EmailTemplateConfig>() ?? new();
+            this.delegationInviteConfig = configuration.GetSection(DelegationInviteConfig.ConfigurationSectionKey).Get<DelegationInviteConfig>() ?? new();
+        }
+
+        /// <inheritdoc/>
+        public async Task AssociateDelegationAsync(string hdid, AssociateDelegationRequest request, CancellationToken ct = default)
+        {
+            ValidationResult validationResult = await new AssociateDelegationRequestValidator().ValidateAsync(request, ct);
+            this.HandleValidationResult(validationResult);
+
+            Guid delegationId = Guid.Parse(this.dataProtector.Unprotect(request.EncryptedDelegationId));
+            this.logger.LogDebug("Delegation Id: {DelegationId} \n Encrypted Delegation Id: {EncryptedDelegationId}", delegationId, request.EncryptedDelegationId);
+
+            Delegation delegation = await this.delegationDelegate.GetDelegationAsync(delegationId) ?? throw new ProblemDetailsException(
+                ExceptionUtility.CreateProblemDetails(ErrorMessages.DelegationNotFound, HttpStatusCode.NotFound, nameof(DelegationService)));
+
+            TimeZoneInfo localTimezone = DateFormatter.GetLocalTimeZone(this.configuration);
+            DateTime referenceDate = TimeZoneInfo.ConvertTimeFromUtc(DateTime.UtcNow, localTimezone);
+            TimeSpan timeDifference = referenceDate - delegation.CreatedDateTime;
+
+            bool validSharingLink = Math.Abs(timeDifference.TotalHours) <= this.delegationInviteConfig.ExpiryHours;
```

Comment:
This seems to be a duplicate validation and is handled by `SharingLinkExpirationValidator`. The expiry link validator is kicked off by the call to the `AssociateDelegationValidator` called in line 111

This variable also doesn't seem to be used anywhere